framework/CMakeLists.txt

```diff
@@ -450,70 +463,78 @@ endif()
 add_library(${PROJECT_NAME} OBJECT ${PROJECT_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
+set(PUBLIC_COMPILE_DEFINITIONS)
+set(PUBLIC_COMPILE_OPTIONS)
+set(PRIVATE_COMPILE_OPTIONS)
+
 # compiler flags based on compiler type
 if(NOT MSVC)
-    target_compile_options(${PROJECT_NAME} PUBLIC -fexceptions)
+    list(APPEND PUBLIC_COMPILE_OPTIONS -fexceptions)
 endif()
 
 if(MSVC)
-    target_compile_options(${PROJECT_NAME} PUBLIC /MP)
+    list(APPEND PUBLIC_COMPILE_OPTIONS /MP)
 endif()
 
 if(${VKB_VALIDATION_LAYERS})
-    target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VALIDATION_LAYERS)
+    list(APPEND PUBLIC_COMPILE_DEFINITIONS VKB_VALIDATION_LAYERS)
 endif()
 
 # GPU assisted validation layers are not available on macOS.
 if(${VKB_VALIDATION_LAYERS_GPU_ASSISTED})
     if (APPLE)
         message(WARNING "GPU assisted validation layers are not currently available on macOS.")
     else()
-        target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VALIDATION_LAYERS)
-        target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VALIDATION_LAYERS_GPU_ASSISTED)
+        list(APPEND PUBLIC_COMPILE_DEFINITIONS
+            VKB_VALIDATION_LAYERS_GPU_ASSISTED)
     endif()
 endif()
 
 if(${VKB_VALIDATION_LAYERS_BEST_PRACTICES})
-    target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VALIDATION_LAYERS)
-    target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VALIDATION_LAYERS_BEST_PRACTICES)
+    list(APPEND PUBLIC_COMPILE_DEFINITIONS
+        VKB_VALIDATION_LAYERS_BEST_PRACTICES)
 endif()
 
 if(${VKB_VALIDATION_LAYERS_SYNCHRONIZATION})
-    target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VALIDATION_LAYERS)
-    target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VALIDATION_LAYERS_SYNCHRONIZATION)
+    list(APPEND PUBLIC_COMPILE_DEFINITIONS
+        VKB_VALIDATION_LAYERS_SYNCHRONIZATION)
 endif()
 
 if(${VKB_VULKAN_DEBUG})
-    target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_VULKAN_DEBUG)
+    list(APPEND PUBLIC_COMPILE_DEFINITIONS VKB_VULKAN_DEBUG)
 endif()
 
 if(${VKB_ENABLE_PORTABILITY})
     message(STATUS "Vulkan Portability extension is enabled")
-    target_compile_definitions(${PROJECT_NAME} PUBLIC VKB_ENABLE_PORTABILITY)
+    list(APPEND PUBLIC_COMPILE_DEFINITIONS VKB_ENABLE_PORTABILITY)
 endif()
 
 if(${VKB_WARNINGS_AS_ERRORS})
     message(STATUS "Warnings as Errors Enabled")
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-        target_compile_options(${PROJECT_NAME} PRIVATE -Werror)
+        list(APPEND PRIVATE_COMPILE_OPTIONS -Werror)
     elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-        target_compile_options(${PROJECT_NAME} PRIVATE /W3 /WX)
+        list(APPEND PRIVATE_COMPILE_OPTIONS /W3 /WX)
     endif()
 endif()
 
+target_compile_definitions(${PROJECT_NAME} PUBLIC ${PUBLIC_COMPILE_DEFINITIONS})
+target_compile_options(${PROJECT_NAME} PUBLIC ${PUBLIC_COMPILE_OPTIONS})
```

Comment:
These are superfluous.  If you're setting public compile options and defines, they will automatically be applied to `framework`, because being public means they're transitive to all downstream consumers.